### PR TITLE
Publish in DBS evt per lumi

### PR DESCRIPTION
**Pull request by dciangot**

### CODE REVIEW

**Comment by emaszs** [comment]
Something to keep in mind is that we still need to do the check, https://github.com/dmwm/AsyncStageout/issues/4542, can't really roll out this feature otherwise I don't think.

**Comment by belforte** [comment]
about the check for "consistent use of eve/lumi in one block" i.e. #4542 
I had a chat with @dciangot about it. I believe there are two sides to this:
1. when assembling the list of files for a block to publish in ASO, ASO code will check that they have
 same kind (in the above respect) and if not, will remove event/lumi for the full block. A line could be logged. But no error raise and no feedback to user
2. since "corrupted" data in DBS is bad and can impact a lot of things, we follow the advice from Yuyi in #4542 and are very paranoid about this, buy having a final check right before calling the DBS API and if something is not correct, raise a fatal exception and abort publication for this dataset.

Idea is that one can always have a subtle bug in 1., we do not want to discover it too late.

File: src/python/AsyncStageOut/PublisherWorker.py [hunk 1]
```diff
@@ -836,8 +835,12 @@ def format_file_3(self, file):
              }
         file_lumi_list = []
         for run, lumis in file['runlumi'].items():
```
Comment (belforte) [inline src/python/AsyncStageOut/PublisherWorker.py]: this should be using iteritems, not items. For efficiency.

File: src/python/AsyncStageOut/PublisherWorker.py [hunk 2]
```diff
@@ -836,8 +835,12 @@ def format_file_3(self, file):
...
-            for lumi in lumis:
-                file_lumi_list.append({'lumi_section_num': int(lumi), 'run_num': int(run)})
+            for lumi, evts in lumis.items():
+                self.logger.debug("lumi: %s, evts: %s" % (lumi,evts))
+                if not evts in ['None']:
```
Comment (belforte) [inline src/python/AsyncStageOut/PublisherWorker.py]: is this a complicated way to write : if evts is not 'None'  ?

Comment (dciangot) [inline src/python/AsyncStageOut/PublisherWorker.py]: You're right, during the test there were 2 values in the list that's the reason. Now it makes no sense other than be easier to add another value ^^

File: src/python/AsyncStageOut/PublisherWorker.py [hunk 3]
```diff
@@ -835,9 +836,23 @@ def format_file_3(self, file):
...
-                file_lumi_list.append({'lumi_section_num': int(lumi), 'run_num': int(run)})
+
+        # check if block contains mixed evt/noevt
+        if self.evtLumi:
+            for _, lumis in file['runlumi'].iteritems():
+                for _, evt in lumis.iteritems():
+                   if evt == 'None':
+                       self.evtLumi = False
+                       break
+	    self.logger.debug("Check on the presence of evts/lumi: %s" % self.evtLumi)
```
Comment (mmascher) [inline src/python/AsyncStageOut/PublisherWorker.py]: this will always be printed since it is inside the if. Should it be put outside it?

Comment (mmascher) [inline src/python/AsyncStageOut/PublisherWorker.py]: Scratch that, I misunderstood

**Pull request merged**